cpp/include/ucxx/delayed_submission.h

```diff
@@ -56,15 +57,135 @@ class DelayedSubmission {
                     const ucs_memory_type_t memoryType = UCS_MEMORY_TYPE_UNKNOWN);
 };
 
+template <typename T>
+class DelayedSubmissionCollectionTemplateSafe {
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**nitpick (non-blocking):** ‏I'm not sure this name is the best. The fact it's templated is given by the type. And it's not really clear what "safe" means in this context. How about `BaseDelayedSubmissionCollection`?